### DISCUSSION TITLE
Have a problem when call ooptimizer_step() in pytorch_lightning

### DISCUSSION THREAD

**Discussion by kobrafarshidi** [Lightning-AI/pytorch-lightning#18200] [code help: CV]
I should make a model with pytorch_lightingin so I have a part for managing optimization so I call  below function 
```
  def optimizer_step(self, epoch_nb, batch_nb, optimizer, optimizer_i, opt_closure = None, on_tpu=False,
    using_native_amp=False, using_lbfgs=False):
        if self.trainer.global_step < 500:
            lr_scale = min(1., float(self.trainer.global_step + 1) / 500.)
            for pg in optimizer.param_groups:
                pg['lr'] = lr_scale * self.hparams.learning_rate

        ## Linear Decay
        else:
            for pg in optimizer.param_groups:
                pg['lr'] = polynomial(self.hparams.learning_rate, self.trainer.global_step, max_iter = self.max_steps)

        optimizer.step(opt_closure)
        optimizer.zero_grad()
```

This code runed without any error in previous version but now I have error end of the discussion. I changed this code with following code but I have bellow error again.

        if self.trainer.global_step < 500:
            lr_scale = min(1., float(self.trainer.global_step + 1) / 500.)
            for pg in optimizer.param_groups:
                pg['lr'] = lr_scale * self.hparams.learning_rate

        ## Linear Decay
        else:
            for pg in optimizer.param_groups:
                pg['lr'] = polynomial(self.hparams.learning_rate, self.trainer.global_step, max_iter = self.max_steps)

        optimizer.step(optimizer_closure)
        optimizer.zero_grad()

but when I run 

    trainer.fit(model,dl)

 I encounter following error 

    MisconfigurationException                 Traceback (most recent call last)
    <ipython-input-46-324898bb4d9a> in <cell line: 1>()
    ----> 1 trainer.fit(model,datamodule)

    10 frames
    /usr/local/lib/python3.10/dist-packages/pytorch_lightning/loops/optimization/closure.py in consume_result(self)
         49         """
         50         if self._result is None:
    ---> 51             raise MisconfigurationException(
         52                 "The closure hasn't been executed."
         53                 " HINT: did you call `optimizer_closure()` in your `optimizer_step` hook? It could also happen because"

    MisconfigurationException: The closure hasn't been executed. HINT: did you call `optimizer_closure()` in your `optimizer_step` hook? It could also happen because the `optimizer.step(optimizer_closure)` call did not execute it internally.

in addition I change `optimizer_closure` without `None` but I have problem yet
I change `optimizer.step(optimizer_closure)` without any parameter `optimizer.step()` have error again

**Comment by foresee-ai** [comment]
I also face the same problem.

**Reply by Jia-Baos** [reply]
Meet the same problem, have you found a solution?
